_posts/2019-06-21-my-infatuation-with-elm.markdown

```diff
@@ -0,0 +1,76 @@
+But over time and many compiler errors, things started to click in ways I never really thought would be possible.
+As a full-stack engineer, Elm makes me want to shift and become a front-end engineer full time.
+Enough about me, let's get started!
+
+## Tooling
+This is a section that I could just write a whole blog about.
+The tooling is one of my favorite things about Elm.
+I'll start with the formatter, [Elm Format]().
+This formatter is uniquely opinionated and not very customizable.
+One of the frustrating parts, when I say **uniquely** opinionated, is that it will allow you to have a line that extends past your desired buffer length.
+Whenever I walk into a formatted Elm file, I feel like I am relaxing in a sparse forest with plenty of fresh air.
+It has a few quirks like always having 2 new lines between any two blocks of code and it tends to freak out when comments are in the middle of functions.
+To me those things are minor in comparison to the ability the formatter gives me to write sloppy code, then poof it is magically beautiful.
+Another amazing thing it does is alphabetizes the import list which makes finding imports so easy a caveman could do it.
+The formatter also allows beginners to come in and somewhat read what is happening in the code.
+
+Another aspect of the tooling that I am a fan of is the tool called [Elm Analyse](https://stil4m.github.io/elm-analyse/).
+This tool is used to lint the Elm files.
+It tells you about unused imports and variables, as well as whether or not all your files are formatted.
+Elm Analyse is fairly customizable, meaning you can turn off and on various features.
+Another feature that I haven't personally used is that it can detect outdated dependencies.
+This is helpful to keep up to date with package changes without having to check Elm Package.
+I'm thankful that this tool has allowed our code base to stay consistent and keep out minor syntactic garbage.
+
+The [Elm Package](https://package.elm-lang.org/) manager is also another aspect I enjoy about Elm because it is easy to read and understand.
+It allows you to search for packages and presents the results in a useful format.
+This allows me to find the functions I need to solve most of my needs.
+Like most package managers it links well to the related functions and types.
+I do wish that Elm Package allowed me to insert a type signature and return related functions with that type signature, like Haskell, but oh well.
+The previous two tooling points are by far my favorite, but I do appreciate a nice package manager and Elm's easy of versioning.
+Publishing a personal package is another story that I'll touch on later, but for now let us move on.
+
+## Compiler
+
+I **LOVE** Elm's compiler.
+It's instruction is that of a caring mother who just wants you to succeed even when you try and cause chaos through your code.
```

Comment:
```suggestion
It's instruction is like that of a caring mother who just wants you to succeed even when you try and cause chaos through your code.
```
Adding `like` makes the sentence a little more readable imo, but it's totally your call.